external/avro/src/main/scala/org/apache/spark/sql/avro/AvroSerializer.scala

```diff
@@ -41,7 +42,7 @@ class AvroSerializer(rootCatalystType: DataType, rootAvroType: Schema, nullable:
   }
 
   private val converter: Any => Any = {
-    val actualAvroType = resolveNullableType(rootAvroType, nullable)
+    val actualAvroType = resolveNullableType(rootAvroType, rootCatalystType, nullable)
```

Comment:
Is the change here related to specifying schema on write?

Reply:
Yes.  I need the original `rootCatalystType` so I can determine how those types map back to the specified schema.  Simply having `nullable` as context is not sufficient :(